ptychodus/controller/reconstructor.py

```diff
@@ -208,14 +209,14 @@ def _redrawPlot(self) -> None:
         ax.set_ylabel(axisY.label)
         ax.grid(True)
 
-        if len(axisX.series) == len(axisY.series):
-            for sx, sy in zip(axisX.series, axisY.series):
-                ax.plot(sx.values, sy.values, '.-', label=sy.label, linewidth=1.5)
-        elif len(axisX.series) == 1:
-            sx = axisX.series[0]
-
-            for sy in axisY.series:
+        if (
+            (len(axisX.series) == len(axisY.series)) or
+            (len(axisX.series) == 1 and len(axisY.series) >= 1)
+        ):
+            for sx, sy in zip(itertools.cycle(axisX.series), axisY.series):
                 ax.plot(sx.values, sy.values, '.-', label=sy.label, linewidth=1.5)
+                if hasattr(sy, 'hi') and hasattr(sy, 'lo'):
+                    ax.fill_between(sx.values, sy.lo, sy.hi, alpha=0.2)
         else:
             logger.error('Failed to broadcast plot series!')
```

Comment:
This section is the most confusing change because two things are happening:

1. A conditional is added which calls fill_between to draw confidence region around the line if the series has lo,hi values
2. The if-elseif-else tree is compacted to reduce code. By using itertools.cycle(), zip() can handle the case when there is only one X-series for multiple Y-ranges. This means all valid plots follow the same code path!